nanachan/extensions/profiles.py

```diff
@@ -119,229 +104,328 @@ async def promo(self, ctx: Context):
 
-        await send_func(embed=embed)
+        return embed
 
     @nana_command()
     @legacy_command()
     async def register(self, ctx: LegacyCommandContext):
-        """ Register yourself into Nana-chan """
+        """Register yourself into Nana-chan"""
         registrars = sorted(self.registrars.items(), key=itemgetter(0))
 
         desc = []
         view = LockedView(self.bot, ctx.author)
         for name, callback in registrars:
             emoji = callback.__self__.__class__.emoji  # type: ignore
-            desc.append(f"{emoji} **{name}**\n{callback.__doc__}")
+            desc.append(f'{emoji} **{name}**\n{callback.__doc__}')
             button = Button(label=name, emoji=emoji)
             button.callback = callback
             view.add_item(button)
 
-        embed = Embed(title=f"{self.emoji} Register",
-                      description='\n\n'.join(desc))
-        embed.set_author(name=ctx.author,
-                         icon_url=ctx.author.display_avatar.url)
+        embed = Embed(title=f'{self.emoji} Register', description='\n\n'.join(desc))
+        embed.set_author(name=ctx.author, icon_url=ctx.author.display_avatar.url)
         await ctx.reply(
             content=f"React with the {self.bot.get_emoji_str('chousen')} option",
             embed=embed,
-            view=view)
+            view=view,
+        )
+
+    @nana_command(description="Edit your own profile.")
+    @legacy_command()
+    async def iam(self, ctx: LegacyCommandContext):
+        assert ctx.guild
+        member = ctx.guild.get_member(ctx.author.id)
+        profile = await self._get_profile(ctx.author.id)
+        assert member
+        embed = self.create_vcard(member, profile)
+        await ctx.send(embed=embed, view=ProfileCreateOrChangeView(self.bot, member, profile))
+
+    @nana_command(description="Display other user's profile.")
+    @legacy_command()
+    async def whois(self, ctx: LegacyCommandContext, other: discord.User):
+        profile_resp = await get_nanapi().user.user_get_profile(other.id)
+        if not success(profile_resp):
+            match profile_resp.code:
+                case 404:
+                    raise CommandError("User has no registered profile.")
+                case _:
+                    raise RuntimeError(profile_resp.result)
+        profile = profile_resp.result
+        assert ctx.guild
+        member = ctx.guild.get_member(other.id)
+        assert member
+        await ctx.send(embed=self.create_vcard(member, profile))
+
+
+class ProfileModal(ui.Modal):
+    def __init__(self, *, title: str, profile: ProfileSearchResult):
+        super().__init__(title=title)
+        self.profile_dict = {
+            'birthday': profile.birthday.date().strftime(r'%Y-%m-%d') if profile.birthday else '',
+            'full_name': profile.full_name if profile.full_name else '',
+            'graduation_year': re.findall(r'\d{4}', profile.graduation_year)[0]
```

Comment:
use `re.compile` or `str.isdigit`. if the expression is longer than a line add an assignment before the dict, looks bad when the if is stretched over several lines in the dict.

Reply:
For this line specifically it was before I decided to split the old `promotion` property so there's no need to check for the digits right now. I will remove it.